server/src/main/java/org/elasticsearch/search/aggregations/Aggregator.java

```diff
@@ -131,7 +132,7 @@ public ParseField parseField() {
         public static SubAggCollectionMode parse(String value) {
             SubAggCollectionMode[] modes = SubAggCollectionMode.values();
             for (SubAggCollectionMode mode : modes) {
-                if (mode.parseField.match(value)) {
+                if (mode.parseField.match(value, LoggingDeprecationHandler.INSTANCE)) {
```

Comment:
I think it'd be cleaner to take the `DeprecationHandler` as an argument here. I really want to minimize the places where we use the static instance of `LoggingDeprecationHandler`. My ulterior motive is that we *might* be able to remove it entirely at some point which would be super neat.